### ISSUE TITLE
Add dockerfile for deployment of next-forge for self-hosting

### ISSUE THREAD

**Issue by arunselvakumar** [haydenbleasel/next-forge#179]

**Comment by haydenbleasel** [comment]
I'm not a Docker user but I'd be open to a PR that implements this!

**Comment by TharinduX** [comment]
This would be great. I'm planing to host this on coolify but currently i'm gettins some issues with nixpacks. +1

**Comment by oldbettie** [comment]
This would be great especially if we could set it up for each package seperately to allow us to integrate each service seperately or as a group

**Comment by coolchigi** [comment]
hi there👋! I'd like to work on this

**Comment by haydenbleasel** [comment]
@coolchigi Go for it!

**Comment by Synatokok** [comment]
Hopefully this is still being looked into. Was hoping to host it on coolify as well

**Comment by coolchigi** [comment]
I was pre-occupied with some other life events but I have some bandwidth now. I actually have a couple of questions before proceeding. I don't know if I should open up a discussion/new pr so I'll just put it here.

TLDR: Should the Docker setup stay within `next-forge init`, or should we create a separate command, e.g., next-forge add docker?

### What I've Done So Far
- Added a templates directory with Docker-related files:
   - Dockerfile.template
   - .dockerignore.template
   - docker-compose.yml.template
...other files

- Integrated Docker setup into the next-forge init process:

```typescript
const setupDocker = async () => {
  const templateDir = join(process.cwd(), 'templates', 'docker');
  const files = [
    { source: 'Dockerfile.template', target: 'Dockerfile' },
    { source: '.dockerignore.template', target: '.dockerignore' },
    { source: 'docker-compose.yml.template', target: 'docker-compose.yml' },
  ];

  // Copy Docker files from templates to new project
  for (const { source, target } of files) {
    await copyFile(join(templateDir, source), target);
  }
  // ...other logic
};

```

### Issues Discovered
When running `next-forge init`, the entire repository gets cloned including the templates directory. This means implementation details (templates) end up in user projects and users get unnecessary files in their codebase.

### Questions Before Proceeding:

Should the Docker setup stay within next-forge init, or should we create a separate command, e.g., next-forge add docker? If we do decide to let the Docker setup stay within next-forge init, is there a better approach to what I did?
